File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/Commit.java
package com.tencent.devops.scm.api.pojo;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 代表仓库提交
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Commit {

    private String sha;
    private String message;
    private Signature author;
    private Signature committer;
    private LocalDateTime commitTime;
    private String link;
    private List<String> added;
    private List<String> modified;
    private List<String> removed;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/Commit.kt
package com.tencent.devops.scm.api.pojo

import java.time.LocalDateTime

data class Commit(
    val sha: String,
    var message: String,
    var author: Signature? = null,
    val committer: Signature? = null,
    val commitTime: LocalDateTime? = null,
    val link: String = "",
    val added: List<String> = listOf(),
    val modified: List<String> = listOf(),
    val removed: List<String> = listOf()
)
